server/sql.go

```diff
@@ -297,168 +297,176 @@ func (s *Server) syncEngineVersionAndSchema(ctx context.Context, instance *api.I
 		}
+	}
 
-		// Upsert user found in the instance
-		for _, user := range instanceMeta.UserList {
-			userUpsert := &api.InstanceUserUpsert{
-				CreatorID:  api.SystemBotID,
-				InstanceID: instance.ID,
-				Name:       user.Name,
-				Grant:      user.Grant,
-			}
-			_, err := s.store.UpsertInstanceUser(ctx, userUpsert)
-			if err != nil {
-				return fmt.Errorf("failed to sync user for instance: %s. Failed to upsert user. Error %w", instance.Name, err)
+	// Delete user no longer found in the instance
+	for _, user := range instanceUserList {
+		found := false
+		for _, dbUser := range instanceMeta.UserList {
+			if user.Name == dbUser.Name {
+				found = true
+				break
 			}
 		}
 
-		// Delete user no longer found in the instance
-		for _, user := range instanceUserList {
-			found := false
-			for _, dbUser := range instanceMeta.UserList {
-				if user.Name == dbUser.Name {
-					found = true
-					break
-				}
+		if !found {
+			userDelete := &api.InstanceUserDelete{
+				ID: user.ID,
 			}
-
-			if !found {
-				userDelete := &api.InstanceUserDelete{
-					ID: user.ID,
-				}
-				err := s.store.DeleteInstanceUser(ctx, userDelete)
-				if err != nil {
-					return fmt.Errorf("failed to sync user for instance: %s. Failed to delete user: %s. Error %w", instance.Name, user.Name, err)
-				}
+			err := s.store.DeleteInstanceUser(ctx, userDelete)
+			if err != nil {
+				return nil, fmt.Errorf("failed to sync user for instance: %s. Failed to delete user: %s. Error %w", instance.Name, user.Name, err)
 			}
 		}
+	}
 
-		// Sync all database metadata first without diving into the deep structure of each database.
-		// Compare the stored db info with the just synced db schema.
-		// Case 1: If item appears in both stored db info and the synced db metadata, then it's a no-op. We rely on syncDatabaseSchema() later to sync its details.
-		// Case 2: If item only appears in the synced schema and not in the stored db, then we CREATE the database record in the stored db.
-		// Case 3: Conversely, if item only appears in the stored db, but not in the synced schema, then we MARK the record as NOT_FOUND.
-		//   	   We don't delete the entry because:
-		//   	   1. This entry has already been associated with other entities, we can't simply delete it.
-		//   	   2. The deletion in the schema might be a mistake, so it's better to surface as NOT_FOUND to let user review it.
-		databaseFind := &api.DatabaseFind{
-			InstanceID: &instance.ID,
-		}
-		dbList, err := s.store.FindDatabase(ctx, databaseFind)
-		if err != nil {
-			return fmt.Errorf("failed to sync database for instance: %s. Failed to find database list. Error %w", instance.Name, err)
-		}
-		for _, databaseMetadata := range instanceMeta.DatabaseList {
-			databaseName := databaseMetadata.Name
+	// Sync all database metadata first without diving into the deep structure of each database.
```

Comment:
This sentence can be used as the method comment to explain the sync strategy.

Reply:
Done.